### PR TITLE
Cleanup and fix management commands.

### PR DESCRIPTION
**Pull request by vil-s**
- Ensure that new switch is not created unless the --create flag is passed.
- Add tests to ensure no regression for the --create fix on all commands.
- Add separate positional arguments for commands (instead of "positionals") to improve the --help documentation.

### CODE REVIEW

File: waffle/management/commands/waffle_flag.py
Comment (clintonb) [inline waffle/management/commands/waffle_flag.py]: Single quotes

File: waffle/management/commands/waffle_flag.py
Comment (clintonb) [inline waffle/management/commands/waffle_flag.py]: Single quotes

File: waffle/management/commands/waffle_flag.py
Comment (clintonb) [inline waffle/management/commands/waffle_flag.py]: Single quotes

File: waffle/management/commands/waffle_sample.py
Comment (clintonb) [inline waffle/management/commands/waffle_sample.py]: Single quotes. I find "does not" to be more understandable (esp. for non-native English speakers) than "doesn't".

File: waffle/tests/test_management.py
Comment (clintonb) [inline waffle/tests/test_management.py]: Applicable to all of your new tests:

1. Use single quotes to maintain consistency.
2. Add an assertion that actually ensures the flag/switch/sample does not exist.

**Review by clintonb — CHANGES_REQUESTED** [review]
Please squash your commits.

**Comment by vil-s** [comment]
Review changes implemented

File: waffle/tests/test_management.py
Comment (clintonb) [inline waffle/tests/test_management.py]: The assertions should come _after_ the calls to the management command. The purpose of the test is to ensure the command doesn't create a new instance.

**Review by clintonb — CHANGES_REQUESTED** [review]
Correct the tests.

File: waffle/tests/test_management.py
Comment (vil-s) [reply waffle/tests/test_management.py]: aha, I misunderstood your previous comment

File: waffle/tests/test_management.py
Comment (jsocol) [reply waffle/tests/test_management.py]: Fwiw, I strongly prefer double quotes when the string has apostrophes in it, to avoid awkward escape sequences:

```py
'don\'t do this if it can be avoided'
"this's better"
'but this is best'
```

**Comment by codeadict** [comment]
This will be a very nice to have.

**Comment by vil-s** [comment]
@clintonb I fixed the tests and fixed the merge conflicts. It should be ready to go now.

**Review by clintonb — APPROVED** [review]

**Comment by clintonb** [comment]
@unklphil please rebase on the master branch, rather than merge master into your own branch. Once that is done, I'll merge.

**Comment by aehlke** [comment]
Just curious why rebasing/squashing in PRs is critical now that github allows squashing as an option in the merge button?

**Comment by clintonb** [comment]
This project using the "Rebase and merge" option solely. That decision for this project wasn't mine, but I agree with it. Merging introduces merge bubbles which are mostly noise. Squashing assumes that every PR should be squashed into a single commit when. The decision to use a single commit is not always applicable. If I happen to fix two separate bugs, they should be two separate commits.

In this case, since "Rebase and merge" is the only option available to me, the branch needs to be properly rebased. GitHub will not allow the PR to be merged without a rebase.

screen shot 2017-10-24 at 11 09 58 pm

**Comment by aehlke** [comment]
Thank you, I appreciate the detailed response - I've wondered about this, that makes sense now!

**Comment by vil-s** [comment]
I used the wrong damned option for resolving management.commands.waffle_switch after squashing and rebasing (I was certain that "use mine" would use mine, but it didn't) and did `git push --force`, and now those changes are lost.

I really don't feel like fixing it, so I'm going to close the issue this PR because it's too much of a hassle for me.

**Comment by jsocol** [comment]
Ugh, I'm sorry, that's the worst!

If you haven't deleted the local checkout, it may be possible to recover, if you want—or for a future situation—via the `git reflog`. It can fix things like accidental rebases, recover things that have been forced-pushed over, etc. There be dragons but it's a very powerful tool.

**Comment by vil-s** [comment]
I'll check tomorrow if it's salvagable

**Pull request closed without merging**
